packages/expo-audio/src/index.ts

```diff
@@ -15,31 +15,18 @@ import { AudioPlayer, AudioRecorder } from './AudioModule.types';
 import { createRecordingOptions } from './utils/options';
 import { resolveSource } from './utils/resolveSource';
 
-export function useAudioPlayer(
-  source: AudioSource | string | number | null = null
-): readonly [AudioPlayer, AudioStatus] {
+export function useAudioPlayer(source: AudioSource | string | number | null = null): AudioPlayer {
   const parsedSource = resolveSource(source);
-  const player = useReleasingSharedObject(() => {
-    return new AudioModule.AudioPlayer(parsedSource);
-  }, [JSON.stringify(parsedSource)]);
+  const player = useReleasingSharedObject(
+    () => new AudioModule.AudioPlayer(parsedSource),
+    [JSON.stringify(parsedSource)]
+  );
 
-  const status = useEvent(player, 'onPlaybackStatusUpdate', {
-    id: player.id,
-    currentTime: player.currentTime,
-    status: 'unknown',
-    timeControlStatus: 'unknown',
-    reasonForWaitingToPlay: 'unknown',
-    mute: player.muted,
-    duration: player.duration,
-    playing: player.playing,
-    loop: player.loop,
-    isBuffering: player.isBuffering,
-    isLoaded: player.isLoaded,
-    playbackRate: player.playbackRate,
-    shouldCorrectPitch: player.shouldCorrectPitch,
-  });
+  return player;
+}
 
-  return [player, status] as const;
+export function useAudioPlayerStatus(player: AudioPlayer): AudioStatus {
+  return useEvent(player, 'onPlaybackStatusUpdate', player.currentStatus);
```

Comment:
We should probably use `useMemo` for the initial value with a dependency on the player, since it's blocking the main thread